packages/php/blueprint/src/ZipExportedSchema.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Automattic\WooCommerce\Blueprint;
+
+class ZipExportedSchema {
+	// exported schema from ExportSchema class
+	protected array $schema;
+
+	// Full path for the zip file
+	protected string $destination;
+
+	// base path
+	protected string $dir;
+
+	// a unique dir name for a single session
+	protected string $working_dir;
+
+	protected array $files;
+
+	/**
+	 * @param $schema
+	 * @param $destination
+	 */
+	public function __construct( $schema, $destination = null ) {
+		$this->schema = $schema;
+
+		$this->dir         = $this->get_default_destination_dir();
+		$this->destination = $destination === null ? $this->dir . '/woo-blueprint.zip' : Util::ensure_wp_content_path( $destination );
+		$this->working_dir = $this->dir . '/' . date( 'Ymd' ) . '_' . time();
+
+		if ( ! class_exists( 'PclZip' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/class-pclzip.php';
+		}
+	}
+
+	protected function get_working_dir_path( $filename ) {
+		return $this->working_dir . '/' . $filename;
+	}
+
+	protected function maybe_create_dir( $dir ) {
+		if ( ! is_dir( $dir ) ) {
+			mkdir( $dir, 0777, true );
+		}
+	}
+
+	public function zip() {
+
+		$this->maybe_create_dir( $this->working_dir );
+
+		// create .json file
+		$this->files[] = $this->create_json_schema_file();
+		$this->files   = array_merge( $this->files, $this->add_resource( 'installPlugin', 'plugins' ) );
+		$this->files   = array_merge( $this->files, $this->add_resource( 'installTheme', 'themes' ) );
```

Comment:
Is `installPlugin` and `installTheme` a step? If so, I'd recommend using the static function `get_step_name()`

Reply:
Updated in [Remove hardcoded string with get_step_name()](https://github.com/woocommerce/woocommerce/pull/49763/commits/bcb6cbc60662d10492d29321b3dcc196be272c40)